### PR TITLE
Donor Profiles UI is implemented

### LINKED ISSUE #5433: Donor Profile UI should be implemented
## User Story

As a developer, I want the Donor Profile frontend app UI to be implemented so that I can begin exploring how the user experience can be improved.

## Details
High fidelity mock ups for the Donor Profile frontend UI have been created. Based on these mock ups, a dummy UI should be implemented so that as a developer I can begin to gather a feeling for how the donor profile UX will go.

## Additional Context
Figma designs: https://www.figma.com/file/BKr2K0XdDyRRoccJbJKRD7/Epic%3A-Donor-Profile?node-id=397%3A1743

## Acceptance Criteria
-   [x] Donor Profile UI is implemented (with static data)
-   [x] Donor Profile Mobile Navigation UI is implemented
-   [x] Donor Profile hash-based routing is implemented (and supports receipt pages)
-   [x] Donor Profile UI is responsive (as illustrated in Figma designs)

### PR DESCRIPTION
**Pull request by henryholtgeerts**
Resolves #5433 

## Description
This PR introduces front-end rendering logic needed to display tabs and navigate the donro profile application. While there is polish to be done further down the road, this PR lays the groundwork for continued development, while avoiding the potential for UI implementation to balloon into a larger and larger pull request.

## Affects
The PR affects front-end rendering logic of the Donor Profile app. It introduces various default tabs, a navigation system, routing and rudimentary mobile styling.

## Visuals
Screencast:
https://www.loom.com/share/82ecabda71364264a869e4daf6c34be7

## Pre-review Checklist
-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions
1. Create a new page
2. Add the Donor Profile block
3. Save, view the block on the frontend
4. Test out various tabs and links (as shown in the Visuals section).

### CODE REVIEW

**Comment by JasonTheAdams** [comment]
@henryholtgeerts I noticed in the video that you had a long list of donations. When clicking "view receipt" the height of the container dropped, leaving only a sliver up top. I could see the UI going completely out of the view. It seems like it would be nice to have it scroll the UI back into view when the user clicks "view receipt". To avoid it being annoying, we may need to check to see if the top of the UI is outside of the viewport before scrolling the user, otherwise, if they click the item at the top of the list and it scrolls them down that could be strange.

cc: @DevinWalker

**Comment by henryholtgeerts** [comment]
@JasonTheAdams This is a good point, and something I was thinking about as well. I anticipate that as I flesh out these screens, the height will be more consistent (we may also move toward having a min-height set), but certainly the idea of scrolling the user could be especially helpful when being viewed in a mobile context. Definitely something on my radar, and worth exploring!

File: src/DonorProfiles/Model.php [hunk 1]
```diff
@@ -68,5 +68,19 @@ public function loadAssets() {
 			GIVE_VERSION,
 			true
 		);
+
+		wp_enqueue_style(
+			'give-google-font-montserrat',
+			'https://fonts.googleapis.com/css?family=Montserrat:100,100i,200,200i,300,300i,400,400i,500,500i,600,600i,700,700i,800,800i,900,900i&display=swap',
```
Comment (kjohnson) [inline src/DonorProfiles/Model.php:74]: For the sake of page load performance, we probably want to consider trimming down the number of font-weights/styles that are loaded. Last I checked, adding additional weights/styles increase page load times.

File: src/DonorProfiles/resources/js/app/components/app/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import DesktopLayout from '../desktop-layout';
+import MobileLayout from '../mobile-layout';
+import { useWindowSize } from '../../hooks';
+
+const App = () => {
+	const { width } = useWindowSize();
+
+	switch ( true ) {
```
Comment (kjohnson) [inline src/DonorProfiles/resources/js/app/components/app/index.js:8]: With two known options, I would suggest using a simple `if/then/else` statement or ternary.

File: src/DonorProfiles/resources/js/app/components/badge/style.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
+.give-donor-profile-badge {
+	background: #fff;
+	box-shadow: 0 1.60377px 6.41509px rgba(0, 0, 0, 0.11);
```
Comment (kjohnson) [inline src/DonorProfiles/resources/js/app/components/badge/style.scss:3]: Assuming this was pulled straight out of Figma, but I don't think that CSS supports _partial_ pixels. My [research](https://stackoverflow.com/a/4309051) suggests that when using `px` the value is either truncated or rounded.

**Review by kjohnson — APPROVED** [review]
Great use of comments to explain some of the decisions - the "why" really helps when it may not otherwise be obvious.

File: src/DonorProfiles/Model.php (hunk 1 above)
Comment (henryholtgeerts) [reply src/DonorProfiles/Model.php]: Good point - done, ref: https://github.com/impress-org/givewp/pull/5444/commits/4361016d69e5ccabcb3f088c9bab13d007f806ee

File: src/DonorProfiles/resources/js/app/components/app/index.js (hunk 2 above)
Comment (henryholtgeerts) [reply src/DonorProfiles/resources/js/app/components/app/index.js]: Done, ref: https://github.com/impress-org/givewp/pull/5444/commits/a12d68be936c9658110f545e0a177339527fe407

File: src/DonorProfiles/resources/js/app/components/badge/style.scss (hunk 3 above)
Comment (henryholtgeerts) [reply src/DonorProfiles/resources/js/app/components/badge/style.scss]: Done, ref: https://github.com/impress-org/givewp/pull/5444/commits/b41203de3d176c9cea0c1c5650854cb926246197

**Pull request merged**